pipenv/utils/resolver.py

```diff
@@ -552,6 +560,28 @@ def constraint_file(self):
             self._constraint_file = self.prepare_constraint_file()
         return self._constraint_file
 
+    def prepare_default_constraint_file(self):
```

Comment:
This feels like something that could be a standalone utility in dependencies.py that is not method of the class that might be helpful towards future refactors.  something like `def prepare_default_constraint_file(directory, project)`

Reply:
Actually, on second thought, this feels very similar to the method `get_constraints_from_deps` -- Perhaps we could extract out the creation of the tracked tempfile and the writing to it of the constraints out to a shared method that takes the directory and the constraints to be written.  Like `def prepare_default_constraint_file(directory, constraints)`.  It could be re-used by `write_constraint_to_file`

Reply:
Yeah, I will try this. Btw, I might not be able to response to this issue in the next few days. I’m sorry. Will get back on to this as soon as possible.

Reply:
It could also be re-used by `prepare_constraint_file` which has similar logic right now.

Reply:
I wrote standalone utility in dependencies.py, but not sure how to use it in ``prepare_constraint_file``. Maybe my implementation is not good. Please tell me if there are any room for improvement.

Reply:
@dqkqd Thanks, it looks great, basically I think you need to add a third parameter which is the `constraints` and have that computed outside the helper utility file, because that is what is different about the three places that write this file.  I think if we can solve for that, it will be ready for merge :-D

Reply:
Oh -- and please add a news fragment :-)

Reply:
I modified it and add news fragment.